airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -142,3 +143,16 @@ def set_autocommit(self, conn, autocommit: Any) -> None:
 
     def get_autocommit(self, conn):
         return getattr(conn, 'autocommit_mode', False)
+
+    def run(self, sql, autocommit=False, parameters=None):
+        """
+        Snowflake-connector doesn't allow natively the execution of multiple SQL statements in the same
+        call. So for allowing to pass files or strings with several queries this method is coded,
+        that relies on run from DBApiHook
+        """
+        conn = self.get_conn()
+        self.set_autocommit(conn, autocommit)
+
+        queries = [item[0] for item in split_statements(StringIO(sql))]
+        for query in queries:
+            super().run(query, autocommit, parameters)
```

Comment:
the problem with this @JavierLopezT is it will reauthenticate for every statement.

instead you should connect only once.  this way you can reuse temp tables, for example.  but also it will be faster

Reply:
Thanks for the example. However, I don't get the print fetchall part

Reply:
After you execute a statement snowflake returns information

Log should print it